runtime/wasm/src/module/mod.rs

```diff
@@ -878,30 +935,29 @@ impl WasmInstanceContext {
             // result type expected by mappings
             true
         });
-        Ok(self.asc_new(&result))
+        self.asc_new(&result)
     }
 
     /// function ipfs.cat(link: String): Bytes
-    fn ipfs_cat(&mut self, link_ptr: AscPtr<AscString>) -> Result<AscPtr<Uint8Array>, Trap> {
-        if !self.allow_non_determinstic_ipfs {
-            return Err(anyhow::anyhow!(
+    fn ipfs_cat(
+        &mut self,
+        link_ptr: AscPtr<AscString>,
+    ) -> Result<AscPtr<Uint8Array>, HostExportError> {
+        if !self.experimental_features.allow_non_deterministic_ipfs {
+            return Err(HostExportError::Deterministic(anyhow!(
```

Comment:
I wouldn't consider this deterministic because it depends on an env var set by the indexer.

Reply:
I spent a while waffling about how to handle this. We've used the word "determinism" to actually mean multiple conflated things in different contexts. In this case, the error is in fact _attestable_. Within the context of The Network this subgraph should always fail with this error, deterministically. I'd be interested in a longer chat on slack about whether or not we should tease these apart somehow or what.

Reply:
I see your point that in the network the this should never be allowed so it is deterministic. Ok we can keep this as deterministic.